### PR TITLE
Beiwe taskrunner

### PR DESCRIPTION
**Pull request by biblicabeebli**
This code makes the following intentional changes:
- Beiwe requires that output CSVs for its consumption to go into a folder named `daily`. Willow, Sycamore, Oak, and Jasmine have all had modifications to place output into a well-named folder based on the input frequency.
- In order to accommodate the above and make the code DRYer Sycamore, Willow, and Jasmine, needed some refactoring. Willow in particular got a new `log_stats_inner` function that also reduces indentation by.... 4 levels I think?
- Sycamore Defintely had a bug - original code:
```python
     if end_date is None:
        end_date = get_month_from_today()  # this just inserts end_date if it wasn't provided

    # Read, aggregate and clean data
    else:
        # And then the rest of the function was indented here inside the else block.
```
- This merge request includes a fix from `hackdna/fix-logging` because the code previously set/reset the global logging level, which matters on a server.

As far as I know nothing was broken by my changes. My current setup of Beiwe's django test runner with the forest library present appears to find and run the Forest test suite without errors.

I do see https://github.com/onnela-lab/forest/pull/227 and I believe the code in my branch is better suited than existing code for adding additional frequency options.

### CODE REVIEW

**Comment by biblicabeebli** [comment]
mypy and flake8 issues are now addressed, normal tests did pass.

**Review by hackdna — COMMENTED** [review]
Sorry for the delay. The tests are failing on Windows now: https://github.com/onnela-lab/forest/actions/runs/7187576824/job/19575427334#step:10:1109
It looks like this is related to #211. I am trying to figure out the best way forward, suggestions welcome (@clementzach).

**Comment by biblicabeebli** [comment]
Ok, this can sit for a while pending various platform issues, but I do need to reference this commit hash in the beiwe-backend codebase.
I'll do my best to merge in any changes on develop so the merge can happen, if I do any further work it will be on top of this branch.

Regarding the librosa dependency bug, I can be flexible on when that dictates the backend/platforrm changes need to be developed - definitely not before the new year.

**Comment by hackdna** [comment]
@GeorgeEfstathiadis since this PR touches some of your code could you have a look and approve if this all looks good to you?

File: forest/willow/log_stats.py
Comment (hackdna) [inline forest/willow/log_stats.py:460]: We use `logger.info("%s", var)` style in log strings (https://forest.beiwe.org/en/latest/development.html#python)

File: forest/willow/log_stats.py
Comment (hackdna) [inline forest/willow/log_stats.py:472]: We use `logger.info("%s", var)` style in log strings (https://forest.beiwe.org/en/latest/development.html#python)

File: forest/willow/log_stats.py
Comment (hackdna) [inline forest/willow/log_stats.py:529]: We use `logger.info("%s", var)` style in log strings (https://forest.beiwe.org/en/latest/development.html#python)

File: forest/willow/log_stats.py
Comment (hackdna) [inline forest/willow/log_stats.py:497]: We use `logger.info("%s", var)` style in log strings (https://forest.beiwe.org/en/latest/development.html#python)

**Review by hackdna — CHANGES_REQUESTED** [review]
Thank you! Just a couple of minor items to address with logging. I've fixed the mp3 dependency issue in Windows. LGTM otherwise, just waiting for approval from @GeorgeEfstathiadis.

**Review by GeorgeEfstathiadis — APPROVED** [review]
I reviewed jasmine and oak changes and it looks good to me

**Comment by biblicabeebli** [comment]
@hackdna I've shifted the log statements to the logger format as requested.

I'd like to keep the branch name around for anticipated future work related to the beiwe taskrunner, currerntly summarized on https://github.com/onnela-lab/beiwe-discussions/issues/222

**Review by hackdna — APPROVED** [review]
Thank you! We try to keep feature branches as short lived as possible to reduce merge conflicts. I'm going to merge this PR, so please create a new branch off develop for future changes.

**Pull request merged**
